### ISSUE TITLE
[BUG] Combining NonEmptyList and an Enumeratum enum doesn't generate a correct schema

### ISSUE THREAD

**Issue by sebver** [softwaremill/tapir#715]
Tapir version: 0.16.13

Scala version: 2.13.3

**Describe the bug**

Something is going wrong when you have a `NonEmptyList[AnEnumeratumEnum]`. The enum values are not included like when you would use a `List[AnEnumeratumEnum]`.
I couldn't figure out why this is happening, maybe someone more familiar with the auto derivation has an idea?

**How to reproduce?**

I made a reproduction here: https://github.com/sebver/tapir-cats-enumeratum-bug/blob/master/src/main/scala/Main.scala

**Additional information**

The difference for completeness' sake:

```
  WithNel:
    required:
    - testEnums
    type: object
    properties:
      testEnums:
        type: array
        items:
          type: string
        minItems: 1
```
vs
```
    WithList:
      type: object
      properties:
        testEnums:
          type: array
          items:
            type: string
            enum:
            - first
            - second
```

**Comment by adamw** [comment]
Fixed in [0.16.15](https://github.com/softwaremill/tapir/releases/tag/v0.16.15)

**Comment by sebver** [comment]
Just tried it and I can confirm it works now. Thank you!
